src/components/Dashboard/Dashboard.tsx

```diff
@@ -0,0 +1,193 @@
+	status: 'completed' | 'focused' | 'not_done' | null;
+}
+
+interface CategorizedGoal {
+	category: 'savings' | 'expenses' | 'investing' | 'charity';
+	goals: Goal[];
+}
+
+interface GoalsData {
+	message: string;
+	categorizedGoals: CategorizedGoal[];
+}
+
+/* 	function to filter through user goals based on status and category in order to populate the dashboard sections. 
+	Usage eg:
+	{addWidgets(goalsData, 'completed', 'savings')} 
+	*/
+function addWidgets(
+	data: GoalsData,
+	status?:
+		| ('completed' | 'focused' | 'not_done' | null)[]
+		| ('completed' | 'focused' | 'not_done' | null),
+	category?: 'savings' | 'expenses' | 'investing' | 'charity'
+) {
+	if (!data || !data.categorizedGoals) return null;
+	console.log(data);
+	const matchingWidgets = data.categorizedGoals
+		.filter((categorizedGoal) => categorizedGoal.category === category)
+		.flatMap((categorizedGoal) => categorizedGoal.goals)
+		.filter((goal) => {
+			if (Array.isArray(status)) {
+				return status.includes(goal.status);
+			}
+			return goal.status === status;
+		})
+		.map((goal) => (
+			<Widget
+				key={goal.id}
+				category={goal.category}
+				description={goal.description}
+				isCoreTask={goal.is_recurrent}
+			/>
+		));
+
+	return matchingWidgets;
+}
+
+export function DashboardContainer({ children }: DashboardContainerProps) {
+	const sections = React.Children.toArray(children);
+
+	return (
+		<div className='container min-h-screen mx-auto p-4 space-y-8'>
+			{sections.map((section, index) => {
+				if (React.isValidElement<DashboardSectionProps>(section)) {
+					return (
+						<div key={index} className='p-4 rounded-lg'>
+							<div className='flex gap-4 items-center bg-lilac rounded-lg p-4 mb-4'>
+								<div className='flex items-center justify-center w-10 h-10 bg-offWhite rounded-full'>
+									<p className='text-xl font-bold'>{index + 1}</p>
+								</div>
+								<h2 className='text-xl font-bold'>{section.props.title}</h2>
+							</div>
+							<div className='grid grid-cols-1 md:grid-cols-2 lg:grid-cols-4 gap-4'>
+								{section.props.children}
+							</div>
+						</div>
+					);
+				}
+				return null;
+			})}
+		</div>
+	);
+}
+
+export function DashboardSection({ title, children }: DashboardSectionProps) {
+	return (
+		<React.Fragment>
+			{React.Children.map(children, (child) => {
+				if (
+					React.isValidElement<WidgetProps>(child) &&
+					(child.type === WidgetHeader || child.type === WidgetContainer)
+				) {
+					return React.cloneElement(child as ReactElement<WidgetProps>, {
+						category: child.props.category,
+					});
+				}
+				return child;
+			})}
+		</React.Fragment>
+	);
+}
+
+export function DashboardLayout({ userId }: DahsboardProps) {
+	const goalsData = useFetch<GoalsData>(`${baseUrl}/goals/user/${userId}`)
+		.data as GoalsData;
+
+	if (!goalsData) {
+		return (
+			<div className='flex items-center justify-center min-h-screen'>
+				<Loader2 className='w-12 h-12 animate-spin text-lilac' />
```

Comment:
this is very cool! Could we console.log that there's an error, so devs can figure out what's wrong?

Reply:
I've changed this to handle error using in-built handling from useFetch function
```
	const [isLoading, setIsLoading] = useState(true);
	const fetchGoals = useFetch<GoalsData>(`${baseUrl}/goals/user/${userId}`);

	const goalsData = fetchGoals.data as GoalsData;

	useEffect(() => {
		if (fetchGoals.data || fetchGoals.error) {
			setIsLoading(false);
		}
	}, [goalsData, fetchGoals.error]);

	if (isLoading) {
		return (
			<div className='flex items-center justify-center min-h-screen'>
				<Loader2 className='w-12 h-12 animate-spin text-lilac' />
			</div>
		);
	}

	if (fetchGoals.error) {
		return (
			<div className='flex items-center justify-center min-h-screen'>
				<p className='text-red-500'>
					Error loading dashboard data. Please try again later.
				</p>
			</div>
		);
	}
	```